lib/utils.js

```diff
@@ -26,7 +28,93 @@ function queryParameters(queryString) {
   return result;
 }
 
+function isPromise(obj) {
+  return obj instanceof Promise ||
+    !!obj && obj.then && typeof obj.then === 'function';
+}
+
+function isFalsy(val) {
+  return !val ||
+    val === false ||
+    val instanceof Error ||
+    typeof val === 'string';
```

Comment:
Yes, when we're emulating the save hook behaviour.  A string returned indicates a custom error message.